test/compressor_test.go

```diff
@@ -39,6 +40,83 @@ import (
 	testpb "google.golang.org/grpc/interop/grpc_testing"
 )
 
+// TestUnsupportedEncodingResponse validates gRPC status codes for different client-server compression setups
+// ensuring the correct behavior when compression is enabled or disabled on either side.
+func (s) TestUnsupportedEncodingResponse(t *testing.T) {
+	tests := []struct {
+		name           string
+		clientUseNop   bool
+		serverUseNop   bool
```

Comment:
These two names are not very descriptive. How about `clientCompress` and `serverCompress`. The fact that the compressor being used is a no-op compressor is actually an implementation detail and need not be part of the name of this field.